### PR TITLE
SysCall improvements and fix icon not displaying

### PR DESCRIPTION
**Pull request by tambry**

### CODE REVIEW

File: rpcs3/Emu/SysCalls/Modules/cellNetCtl.cpp [hunk 1]
```diff
@@ -48,7 +52,15 @@ int cellNetCtlGetState(vm::ptr<u32> state)
 {
 	cellNetCtl.Warning("cellNetCtlGetState(state_addr=0x%x)", state.addr());
 
-	*state = CELL_NET_CTL_STATE_Disconnected; // TODO: Allow other states
+	// TODO: Unix/Linux and Mac implementation
+#ifdef _WIN32
+	if (InternetCheckConnection(L"http://www.google.com", FLAG_ICC_FORCE_CONNECTION, 0))
```
Comment (Bigpet) [inline rpcs3/Emu/SysCalls/Modules/cellNetCtl.cpp]: This is a really ugly "solution" in my opinion. Also, I don't think whether the host machine is or is not connected to the net should directly effect the emulated system.

Comment (tambry) [inline rpcs3/Emu/SysCalls/Modules/cellNetCtl.cpp]: @Bigpet What do you mean should effect the emulated system (PS3)? It most definently does affect many games, whether you're connected to the Internet or not.

**Comment by Bigpet** [comment]
I don't think adding the *.aps file is necessary. Seems redundant when there's a *.rc file already

File: rpcs3/Emu/SysCalls/Modules/cellNetCtl.cpp (hunk 1 above)
Comment (Bigpet) [inline rpcs3/Emu/SysCalls/Modules/cellNetCtl.cpp]: I missed a word there, I wanted to write "should not directly effect the emulated system"

File: rpcs3/Emu/SysCalls/lv2/sys_tty.cpp [hunk 2]
```diff
@@ -22,7 +22,7 @@ s32 sys_tty_write(s32 ch, vm::ptr<const void> buf, u32 len, vm::ptr<u32> pwritel
 {
 	sys_tty.Log("sys_tty_write(ch=%d, buf_addr=0x%x, len=%d, preadlen_addr=0x%x)", ch, buf.addr(), len, pwritelen.addr());
 
-	if(ch > 15 || (s32)len <= 0) return CELL_EINVAL;
+	if(ch > 15 || len == 0) return CELL_EINVAL;
```
Comment (Nekotekina) [inline rpcs3/Emu/SysCalls/lv2/sys_tty.cpp]: What is the purpose of this change?

Comment (Bigpet) [inline rpcs3/Emu/SysCalls/lv2/sys_tty.cpp]: Well it makes more sense than the old one. If you go back in history you'll see that it was `len<=0` which caused a warning. The warning was "fixed with" `(s32)len <=0` although `len==0` makes more sense.

Comment (Nekotekina) [inline rpcs3/Emu/SysCalls/lv2/sys_tty.cpp]: It made me take a look into LV2. It seems that signed comparison is correct, but it doesn't return EINVAL, just CELL_OK if len is <= 0.

File: rpcs3/Emu/SysCalls/lv2/sys_tty.cpp [hunk 3]
```diff
@@ -22,7 +22,17 @@ s32 sys_tty_write(s32 ch, vm::ptr<const void> buf, u32 len, vm::ptr<u32> pwritel
...
+	{
+		sys_tty.Error("sys_tty_write(): specified channel was higher than 15.");
+		return CELL_OK;
+	}
+
+	if (len == 0)
+	{
+		sys_tty.Error("sys_tty_write(): specified length was 0.");
+		return CELL_OK;
+	}
```
Comment (Nekotekina) [inline rpcs3/Emu/SysCalls/lv2/sys_tty.cpp]: But it's not what I meant...

Comment (tambry) [inline rpcs3/Emu/SysCalls/lv2/sys_tty.cpp]: @Nekotekina What did you mean then?

Comment (Nekotekina) [inline rpcs3/Emu/SysCalls/lv2/sys_tty.cpp]: I don't understand what you didn't understand...
`if (ch > 15) return CELL_EINVAL;` -- This is ok, I mentioned only len check.
`if ((s32)len <= 0) return CELL_OK;` -- This it what actually happens. Even though `len` type is `u32`, it's checked as if it were `s32`.
